### PR TITLE
test: enable e2e test and fix tests

### PR DESCRIPTION
**Pull request by tiborux**
Enable and fix e2e tests

### CODE REVIEW

File: packages/x-components/tests/e2e/specs/cucumber/search-box.spec.ts [hunk 1]
```diff
@@ -69,15 +68,6 @@ And(
 );
 
 // Scenario 2
-And('the number of next query results are stored', () => {
-  if (cy.$$('[data-test = "next-queries"]').length > 0) {
```
Comment (javieri-empathy) [inline packages/x-components/tests/e2e/specs/cucumber/search-box.spec.ts:73]: 4 hours later, I finally managed to test that this implementation for this step looks like it works.

```
  cy.get('body').then($body => {
    nextQueriesResults = $body.find('[data-test="next-query"]').length;
  });
```

Comment (javieri-empathy) [reply packages/x-components/tests/e2e/specs/cucumber/search-box.spec.ts:73]: Which somehow is the [recommended approach](https://docs.cypress.io/guides/core-concepts/conditional-testing#Element-existence) for this cases.

File: packages/x-components/package.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "test:unit-watch": "vue-cli-service test:unit --watch",
     "test:e2e": "vue-cli-service test:e2e",
     "test:e2e:ci": "vue-cli-service test:e2e --headless --browser chrome",
-    "test": "npm run lint && npm run test:unit",
+    "test": "npm run lint && npm run test:unit && npm run test:e2e:ci",
```
Comment (javieri-empathy) [inline packages/x-components/package.json:25]: I just realized we are running the lint 3 times for the x-components package.

1.  As part of the global `npm run lint` command
2. As part of the x-components `test` command
3. As part of the `x-components `npm run test:e2e:ci` command

Remove the `npm run lint` here, and set the `test:e2e:ci` command to `vue-cli-service --skip-plugins eslint test:e2e --headless --browser chrome`

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

**Review by davidmfempathy — CHANGES_REQUESTED** [review]
Nothing else to add to Javier's review.

**Review by javieri-empathy — APPROVED** [review]

**Review by davidmfempathy — APPROVED** [review]

**Pull request merged**